devices/CharacterLcd/LcdInterface.cs

```diff
@@ -0,0 +1,189 @@
+        public abstract void SendData(byte value);
+
+        /// <summary>
+        /// Sends command to the LCD device
+        /// </summary>
+        /// <param name="command">Byte representing the command</param>
+        public abstract void SendCommand(byte command);
+
+        /// <summary>
+        /// Sends data to the LCD device
+        /// </summary>
+        /// <param name="values">Bytes to be send to the device</param>
+        public abstract void SendData(SpanByte values);
+
+        /// <summary>
+        /// Sends data to the LCD device
+        /// </summary>
+        /// <param name="values">Char value to be send to the device</param>
+        public abstract void SendData(SpanChar values);
+
+        /// <summary>
+        /// Send commands to the LCD device
+        /// </summary>
+        /// <param name="values">Each byte represents command to be send</param>
+        public abstract void SendCommands(SpanByte values);
+
+        /// <summary>
+        /// The initialization sequence and some other complex commands should be sent with delays, or the display may
+        /// behave unexpectedly. It may show random, blinking characters
+        /// or display text very faintly only.
+        /// </summary>
+        /// <param name="command">The command to send</param>
+        public virtual void SendCommandAndWait(byte command)
+        {
+            // Must not run the init sequence to fast or undefined behavior may occur
+            SendCommand(command);
+            Thread.Sleep(1);
+        }
+
+        /// <summary>
+        /// True if device uses 8-bits for communication, false if device uses 4-bits
+        /// </summary>
+        public abstract bool EightBitMode { get; }
+
+        /// <summary>
+        /// The command wait time multiplier for the LCD.
+        /// </summary>
+        /// <remarks>
+        /// In order to handle controllers that might be running at a much slower clock
+        /// we're exposing a multiplier for any "hard coded" waits. This can also be
+        /// used to reduce the wait time when the clock runs faster or other overhead
+        /// (time spent in other code) allows for more aggressive timing.
+        ///
+        /// There is a busy signal that can be checked that could make this moot, but
+        /// currently we are unable to check the signal fast enough to make gains (or
+        /// even equal) going off hard timings. The busy signal also requires having a
+        /// r/w pin attached.
+        /// </remarks>
+        public double WaitMultiplier { get; set; } = 1.0;
+
+        /// <summary>
+        /// Wait for the device to not be busy.
+        /// </summary>
+        /// <param name="microseconds">Time to wait if checking busy state isn't possible/practical.</param>
+        public virtual void WaitForNotBusy(int microseconds)
+        {
+            Thread.Sleep(microseconds / 10); //TODO: we should use SpinWait under a certain value...
```

Comment:
Highlighting this as it fixed most current errors I had, although it is probably not the best way!

Reply:
See line below for how it was originally handled.